src/App.js

```diff
@@ -1,16 +1,41 @@
 import React from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+// import ChatEntry from './components/ChatEntry';
+import ChatLog from './components/ChatLog';
+import {useState} from 'react'
 
 const App = () => {
+  
+  const[textMessage, setTextMessage] = useState(chatMessages)
+  const likedMessage = (id) => {
+  const isliked = textMessage.map((text) =>{
+    if (text.id === id){
+      return {...text, liked: !text.liked}
+    } else{
+      return text;
+    }
+  })
+    setTextMessage(isliked);
```

Comment:
👀 Since the next version of the chat data depends on the current version, prefer using the callback style of setChatMessagData:
```javascript
setTextMessage(prevEntries => ( 
    // logic to use current chats to make a new copy with the desired liked status flipped (the map logic above) 
    // return the new chats to update the state 
))
```

You can see an example of how we did it in Flasky [here](https://github.com/Ada-C19/flasky-front-end/blob/zoisite/src/App.js#L79-L84)